messages/src/message.rs

```diff
@@ -679,6 +717,132 @@ impl SerializedSize for Status {
+        /// by the Sidecar board design itself.
+        const ENABLED               = 0b0000_0000_0010;
+
+        /// The module is held in reset.
+        ///
+        /// This translates to the `ResetL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.2.
+        const RESET                 = 0b0000_0000_0100;
+
+        /// The module is held in low-power mode.
+        ///
+        /// This translates to the `ResetL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.3.
+        const LOW_POWER_MODE        = 0b0000_0000_1000;
+
+        /// At least one interrupt is signaled on the module. The details of the
+        /// interrupt cause can be queried by reading various bytes of the QSFP
+        /// memory map, such as bytes 3-21 in the lower memory page. See
+        /// SFF-8636 rev 2.10a section 6.2.3 for details.
+        ///
+        /// This translates to the `IntL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.5.
+        const INTERRUPT             = 0b0000_0001_0000;
+
+        /// This module's power supply has come up successfully.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const POWER_GOOD            = 0b0000_0010_0000;
+
+        /// This module's power supply has not come up after being enabled.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const FAULT_POWER_TIMEOUT   = 0b0000_0100_0000;
+
+        /// This module unexpectedly lost power.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const FAULT_POWER_LOST      = 0b0000_1000_0000;
+
+        /// The SP has disabled the port associated with this module
+        ///
+        /// This occurs if the module was initialized but is now NACKing I2C
+        /// requests for its temperature.
+        const DISABLED_BY_SP      = 0b1_0000_0000;
+    }
+}
+impl fmt::Display for StatusV2 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl core::str::FromStr for StatusV2 {
+    type Err = bitflags::parser::ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.parse().map(Self)
+    }
+}
+
+impl Serialize for StatusV2 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_u32(self.bits())
+    }
+}
+
+struct StatusV2Visitor;
+
+impl<'de> Visitor<'de> for StatusV2Visitor {
+    type Value = StatusV2;
+
+    fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "a single u32")
+    }
+
+    fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+    where
+        E: SerdeError,
+    {
+        u32::try_from(v)
+            .map(StatusV2::from_bits_truncate)
```

Comment:
Is the `from_bits_truncate` call needed here? It could just be `from_bits` since the conversion succeeded in this `map` call, right?